File: 14.longest-common-prefix.ts
/*
 * @lc app=leetcode id=14 lang=typescript
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
function longestCommonPrefix(strs: string[]): string {
    if (strs == null || strs.length == 0) {
        return ""
    }
    let ans: string = ""
    for (let i = 0; i < strs[0].length; i++) {
        let curr: string = strs[0][i]
        for (let str of strs) {
            if (i >= str.length || str[i] != curr) {
                return ans
            }
        }
        ans += curr
    }
    return ans
};
// @lc code=end

File: 14.longest-common-prefix.go
/*
 * @lc app=leetcode id=14 lang=golang
 *
 * [14] Longest Common Prefix
 */

package leetcodesolution

// @lc code=start
func longestCommonPrefix(strs []string) string {
	if strs == nil || len(strs) == 0 {
		return ""
	}
	ans := make([]byte, 0)
	for i := 0; i < len(strs[0]); i++ {
		curr := strs[0][i]
		for _, str := range strs {
			if i >= len(str) || str[i] != curr {
				return string(ans)
			}
		}
		ans = append(ans, curr)
	}
	return string(ans)
}

// @lc code=end
